### ISSUE TITLE
RandomCutMixV2 never draws the last two cut positions per axis, and a full cut starts at -1

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4730]
## Describe the bug

`RandomCutMixV2`'s generator (`kornia/augmentation/random_generator/_2d/cutmix.py`) draws the start of each cut as

```python
x_start = (u * (width - cut_width - 1)).floor()   # u ~ U[0, 1)
```

The legal starts are `0 .. width - cut_width`, i.e. `width - cut_width + 1` positions. The `- 1` plus the `floor` limit the draw to `0 .. width - cut_width - 2`, so on each axis the **last two legal starts are never drawn**. Cuts never touch the last row or column, and the final pixels are pasted less often than a uniform placement would paste them. The same arithmetic applies to `y_start`.

When `cut_size=(0.0, 0.0)` gives a full-image cut (`cut_width == width`), the span becomes `-1`, so the start is `-1` and the box sits one pixel outside the image:

```python
import torch, kornia.augmentation as K
torch.manual_seed(0)
aug = K.RandomCutMixV2(p=1.0, cut_size=(0.0, 0.0))
print(aug.forward_parameters(torch.Size([2, 1, 10, 12]))["crop_src"][0, 0].tolist())
# [[-1.0, -1.0], [10.0, -1.0], [10.0, 8.0], [-1.0, 8.0]]  -> 99 of 120 pixels pasted, while lambda reports a full cut

aug = K.RandomCutMixV2(p=1.0, cut_size=(0.5, 0.5))
edges = []
for s in range(2000):
    torch.manual_seed(s)
    edges.append(aug.forward_parameters(torch.Size([4, 1, 8, 8]))["crop_src"][..., 2, 0].max().item())
print(max(edges))  # 5.0 on an 8-wide image with 4-wide cuts: legal far-corner x values go up to 7
```

The code comment says "Reserve at least 1 pixel for cropping". Neither the reservation nor the extra lost position is documented as a contract, and the original CutMix reference draws the box centre uniformly over the whole image.

## Expected behavior

Draw starts uniformly over `0 .. size - cut_size`, inclusive. For example, use `floor(u * (size - cut_size + 1))`, clamped to `size - cut_size` because of the float32 round-up at `u → 1`. Then a full cut starts at 0.

## Notes

Found while reviewing #4721, which makes the placement per-cut (#4712) and keeps this arithmetic unchanged. A fix changes the seeded CutMix stream. Following the #4610 and #4721 precedent, it belongs in a `fixed` fragment that says so.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
